firebase-firestore/src/commonMain/kotlin/dev/gitlive/firebase/firestore/Timestamp.kt

```diff
@@ -1,9 +1,31 @@
 package dev.gitlive.firebase.firestore
 
+import kotlinx.serialization.Serializable
+
+/** A class representing a platform specific Firebase Timestamp. */
+expect class PlatformTimestamp
+
 /** A class representing a Firebase Timestamp. */
-expect class Timestamp
+@Serializable(with = TimestampSerializer::class)
+expect class Timestamp internal constructor(platformValue: PlatformTimestamp) {
+    constructor(seconds: Long, nanoseconds: Int)
+    val seconds: Long
+    val nanoseconds: Int
+    /**
+     * Whether timestamp represents a server time stamp.
+     * if `true` seconds and nanoseconds do not represent any meaningful data.
+     */
+    val isServerTimestamp: Boolean
```

Comment:
This looks like workaround, when this flag is set not fields are used and it behaves as different object. So this object slip must be a level above.

Reply:
yes, I mentioned it [here](https://github.com/splendo/firebase-kotlin-sdk/pull/45/files#diff-71356e9252a0c6e085a4826fbb8001a0d3c55173b62fd8f8e7822c2ebd945bd0R12)

found a better solution just now, applying...